src/ui/components/ide/monaco-editor.tsx

```diff
@@ -36,13 +37,29 @@ export default function (props: { onChange: OnChange }) {
 		ide: { refreshIDE, syncMonacoModels, getActiveFile },
 	} = useCommands();
 
+	// Add a state for theme (light/dark)
+	const savedTheme = (localStorage.getItem('editorTheme') as 'dark' | 'light') || 'dark';
+	const [currentTheme, setCurrentTheme] = useState<'dark' | 'light'>(savedTheme);
+
+	const toggleTheme = () => {
+		setCurrentTheme((prevTheme) => {
+			const newTheme = prevTheme === 'dark' ? 'light' : 'dark';
+			localStorage.setItem('editorTheme', newTheme); // Save the new theme to localStorage
+			return newTheme;
+		});
+	};
+
 	useEffect(() => {
 		if (!monacoRef) {
 			return;
 		}
 
 		monacoRef.editor.defineTheme('theme', theme as any);
-		monacoRef.editor.setTheme('theme');
+		monacoRef.editor.defineTheme('lightTheme', lightTheme as any);
```

Comment:
Can we try centralising themes? Right now 'lightTheme' is an random string being used, it would be better if this was made a first class citizen.

How about we have a folder structure like:
```
theme-manager/
├─ themes/
│  ├─ dark/
│  │  ├─ theme.json
│  │  ├─ dark-theme.ts
│  ├─ light/
│  │  ├─ theme.json
│  │  ├─ light-theme.ts
│  ├─ cobalt/
│  │  ├─ theme.json
│  │  ├─ cobalt-theme.ts
│  ├─ theme.interface.ts
├─ theme-manager.ts
├─ theme-switcher.ts
```
## Theme Manager
This will be our central API for doing anything theme related, be it fetching, parsing or loading. For now it will be simple, and would have two jobs:
- be the central store for all themes
- help parse theme strings('light' etc) into `Theme` objects.

For example:
```typescript
class ThemeManager {
  private themes: Map<slug, Theme> = new Map()
  .set(lightTheme.slug, lightTheme)
  .set(darkTheme.slug, darkTheme)
  .set(cobaltTheme.slug, cobaltTheme);
  
  parseSlug(slug: string): Theme{
    const theme = this.themes.get(slug);
    if(!theme) {
      //throw error
    }
    
    return error;
  }
}
export new ThemeManager();
```

## Theme Interface
This is the shape that all themes must follow. It would look like:
```typescript
interface Theme {
  slug: string;
  getJson(): Record<string, any>;
}
```

## Theme Switcher
This will be the component that switches between themes. It can be rendered anywhere we want.

## Why all this?
- Having a central setup like this, makes it very easy to add more themes, even dynamically or user created ones!
- It also makes it easy to change something since everything is already in one place.
- And a common interface means we don't have to worry about different themes, they are all `Theme`s to us!

Reply:
Also, rather than `theme.json` we can be more specific by saying `light-theme.json`.